doc/man/man3/NewFileGui.3
.TH "NewFileGui" 3 "Fri Jun 12 2015" "Version 0.0.1" "RIDE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
NewFileGui \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <NewFileGui\&.h>\fP
.PP
Inherits QWidget\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBNewFileGui\fP (QWidget *parent=0)"
.br
.ti -1c
.RI "void \fBsetNorthTabWidgetPtr\fP (QTabWidget *masterTabWidgetPtr)"
.br
.ti -1c
.RI "QTabWidget * \fBgetMasterTabWidgetPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetHighlighterPtr\fP (\fBHighlighter\fP *\fBhighlighterPtr\fP)"
.br
.ti -1c
.RI "\fBHighlighter\fP * \fBgetHighlighterPtr\fP ()"
.br
.ti -1c
.RI "void \fBinitBtns\fP ()"
.br
.ti -1c
.RI "void \fBswapNextPage\fP ()"
.br
.ti -1c
.RI "void \fBswapBackPage\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_1\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_1\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_2\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_2\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_3\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_3\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_4\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_4\fP ()"
.br
.ti -1c
.RI "QString * \fBtoString\fP ()"
.br
.ti -1c
.RI "\fB~NewFileGui\fP ()"
.br
.in -1c
.SS "Private Types"

.in +1c
.ti -1c
.RI "enum \fBPage\fP { \fBPAGE_ONE\fP, \fBPAGE_TWO\fP, \fBPAGE_THREE\fP, \fBPAGE_FOUR\fP }"
.br
.in -1c
.SS "Private Slots"

.in +1c
.ti -1c
.RI "void \fBhandleBackBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleNextBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleFinishBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleHelpBntSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleCancelBtnSlot\fP ()"
.br
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "QAbstractItemModel * \fBmodelFromFile\fP (const QString &fileName)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "\fBPage\fP \fBcurrentPage\fP"
.br
.ti -1c
.RI "QPushButton * \fBbackBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBnextBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBfinishBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBhelpBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBcancelBtn\fP"
.br
.ti -1c
.RI "QGridLayout * \fBouterLayout\fP"
.br
.ti -1c
.RI "QHBoxLayout * \fBbuttonLayout\fP"
.br
.ti -1c
.RI "QTabWidget * \fBnorthTabWidgetPtr\fP"
.br
.ti -1c
.RI "\fBHighlighter\fP * \fBhighlighterPtr\fP"
.br
.ti -1c
.RI "\fBFileGui\fP * \fBeditor\fP"
.br
.ti -1c
.RI "QCompleter * \fBcompleter\fP"
.br
.ti -1c
.RI "\fBNewFilePage_1\fP * \fBnewFilePage_1Ptr\fP"
.br
.ti -1c
.RI "\fBNewFilePage_2\fP * \fBnewFilePage_2Ptr\fP"
.br
.ti -1c
.RI "\fBNewFilePage_3\fP * \fBnewFilePage_3Ptr\fP"
.br
.ti -1c
.RI "\fBNewFilePage_4\fP * \fBnewFilePage_4Ptr\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 44 of file NewFileGui\&.h\&.
.SH "Member Enumeration Documentation"
.PP 
.SS "enum \fBNewFileGui::Page\fP\fC [private]\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fIPAGE_ONE \fP\fP
.TP
\fB\fIPAGE_TWO \fP\fP
.TP
\fB\fIPAGE_THREE \fP\fP
.TP
\fB\fIPAGE_FOUR \fP\fP
.PP
Definition at line 49 of file NewFileGui\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "NewFileGui::NewFileGui (QWidget *parent = \fC0\fP)"

.PP
Definition at line 5 of file NewFileGui\&.cpp\&.
.SS "NewFileGui::~NewFileGui ()"

.PP
Definition at line 332 of file NewFileGui\&.cpp\&.
.SH "Member Function Documentation"
.PP 
.SS "\fBHighlighter\fP * NewFileGui::getHighlighterPtr ()"

.PP
Definition at line 45 of file NewFileGui\&.cpp\&.
.SS "QTabWidget * NewFileGui::getMasterTabWidgetPtr ()"

.PP
Definition at line 33 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::handleBackBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 51 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::handleCancelBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 135 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::handleFinishBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 63 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::handleHelpBntSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 129 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::handleNextBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 57 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::initBtns ()"

.PP
Definition at line 168 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::loadPage_1 ()"

.PP
Definition at line 252 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::loadPage_2 ()"

.PP
Definition at line 268 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::loadPage_3 ()"

.PP
Definition at line 284 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::loadPage_4 ()"

.PP
Definition at line 300 of file NewFileGui\&.cpp\&.
.SS "QAbstractItemModel * NewFileGui::modelFromFile (const QString &fileName)\fC [private]\fP"

.PP
Definition at line 141 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::setHighlighterPtr (\fBHighlighter\fP *highlighterPtr)"

.PP
Definition at line 39 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::setNorthTabWidgetPtr (QTabWidget *masterTabWidgetPtr)"

.PP
Definition at line 27 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::swapBackPage ()"

.PP
Definition at line 196 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::swapNextPage ()"

.PP
Definition at line 224 of file NewFileGui\&.cpp\&.
.SS "QString * NewFileGui::toString ()"

.PP
Definition at line 316 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::unloadPage_1 ()"

.PP
Definition at line 260 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::unloadPage_2 ()"

.PP
Definition at line 276 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::unloadPage_3 ()"

.PP
Definition at line 292 of file NewFileGui\&.cpp\&.
.SS "void NewFileGui::unloadPage_4 ()"

.PP
Definition at line 308 of file NewFileGui\&.cpp\&.
.SH "Member Data Documentation"
.PP 
.SS "QPushButton* NewFileGui::backBtn\fC [private]\fP"

.PP
Definition at line 59 of file NewFileGui\&.h\&.
.SS "QHBoxLayout* NewFileGui::buttonLayout\fC [private]\fP"

.PP
Definition at line 66 of file NewFileGui\&.h\&.
.SS "QPushButton* NewFileGui::cancelBtn\fC [private]\fP"

.PP
Definition at line 63 of file NewFileGui\&.h\&.
.SS "QCompleter* NewFileGui::completer\fC [private]\fP"

.PP
Definition at line 71 of file NewFileGui\&.h\&.
.SS "\fBPage\fP NewFileGui::currentPage\fC [private]\fP"

.PP
Definition at line 57 of file NewFileGui\&.h\&.
.SS "\fBFileGui\fP* NewFileGui::editor\fC [private]\fP"

.PP
Definition at line 70 of file NewFileGui\&.h\&.
.SS "QPushButton* NewFileGui::finishBtn\fC [private]\fP"

.PP
Definition at line 61 of file NewFileGui\&.h\&.
.SS "QPushButton* NewFileGui::helpBtn\fC [private]\fP"

.PP
Definition at line 62 of file NewFileGui\&.h\&.
.SS "\fBHighlighter\fP* NewFileGui::highlighterPtr\fC [private]\fP"

.PP
Definition at line 69 of file NewFileGui\&.h\&.
.SS "\fBNewFilePage_1\fP* NewFileGui::newFilePage_1Ptr\fC [private]\fP"

.PP
Definition at line 73 of file NewFileGui\&.h\&.
.SS "\fBNewFilePage_2\fP* NewFileGui::newFilePage_2Ptr\fC [private]\fP"

.PP
Definition at line 74 of file NewFileGui\&.h\&.
.SS "\fBNewFilePage_3\fP* NewFileGui::newFilePage_3Ptr\fC [private]\fP"

.PP
Definition at line 75 of file NewFileGui\&.h\&.
.SS "\fBNewFilePage_4\fP* NewFileGui::newFilePage_4Ptr\fC [private]\fP"

.PP
Definition at line 76 of file NewFileGui\&.h\&.
.SS "QPushButton* NewFileGui::nextBtn\fC [private]\fP"

.PP
Definition at line 60 of file NewFileGui\&.h\&.
.SS "QTabWidget* NewFileGui::northTabWidgetPtr\fC [private]\fP"

.PP
Definition at line 68 of file NewFileGui\&.h\&.
.SS "QGridLayout* NewFileGui::outerLayout\fC [private]\fP"

.PP
Definition at line 65 of file NewFileGui\&.h\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for RIDE from the source code\&.


src/NewFileGui.cpp
#include "NewFileGui.h"
#include "FileTreeGui.h"


NewFileGui::NewFileGui(QWidget* parent/*, QTabWidget* northTabWidgetPtr*/) : QWidget(parent), completer(0)
{   
    this->setWindowIcon(QIcon("/home/james/NetBeansProjects/ride/images/project2.jpg") );
    this->setWindowTitle("Ride");
    
    outerLayout = new QGridLayout(this);
    newFilePage_1Ptr = new NewFilePage_1();
    newFilePage_2Ptr = new NewFilePage_2();
    newFilePage_3Ptr = new NewFilePage_3();
    newFilePage_4Ptr = new NewFilePage_4();
    
    currentPage = PAGE_ONE;
    
    initBtns();
    
    
    
    
    this->setLayout(outerLayout);
}


void NewFileGui::setNorthTabWidgetPtr(QTabWidget* northTabWidgetPtr)
{
    this->northTabWidgetPtr = northTabWidgetPtr;
}


QTabWidget* NewFileGui::getMasterTabWidgetPtr()
{
    return northTabWidgetPtr;
}


void NewFileGui::setHighlighterPtr(Highlighter* highlighter)
{
    this->highlighterPtr = highlighter;
}


Highlighter* NewFileGui::getHighlighterPtr()
{
    return highlighterPtr;
}


void NewFileGui::handleBackBtnSlot()
{
    swapBackPage();
}


void NewFileGui::handleNextBtnSlot()
{
    swapNextPage();
}


void NewFileGui::handleFinishBtnSlot()
{
    cout << "\"Finish\" button selected" << endl;
    newFilePage_1Ptr->triggerMutators();
    cout << "successfully triggered f1" << endl;
    newFilePage_2Ptr->triggerMutators();
    cout << "successfully triggered f2" << endl;
    newFilePage_3Ptr->triggerMutators();
    cout << "successfully triggered f3" << endl;
    //newFilePage_4Ptr updates automatically
    newFilePage_4Ptr->triggerMutators();
    cout << "successfully triggered f4" << endl;
    cout << toString()->toStdString() << endl;
    
    //create file
    //- - - - - - - - - - - - - - - - - - - - - - - - - - -
    QFont font;
    font.setFamily("Monospace");
    font.setFixedPitch(true);
    font.setPointSize(10);
    font.setWeight(10);

    editor = new FileGui();
    editor->setFont(font);
    
    int tabSize = 4;
    QFontMetrics metrics(font);
    editor->setTabStopWidth(tabSize * metrics.width(' '));
    
    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

    completer = new QCompleter(this);
    completer->setModel(modelFromFile("wordlist.txt"));
    completer->setModelSorting(QCompleter::CaseInsensitivelySortedModel);
    completer->setCaseSensitivity(Qt::CaseInsensitive);
    editor->setCompleter(completer);
    
    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    
    setHighlighterPtr(highlighterPtr = new Highlighter(editor->document() ) );

    //create physical file and tab
    RFile* rideFile = new RFile(*newFilePage_4Ptr->getLocStrPtr()
                                    + "/"
                                    + *newFilePage_4Ptr->getFileNameStrPtr()
                                    + *newFilePage_4Ptr->getFileExtStrPtr() );
    //-----------------
    cout << "\t\tLoc: " << newFilePage_4Ptr->getLocStrPtr()->toStdString() << endl;
    cout << "\t\tName: " << newFilePage_4Ptr->getFileNameStrPtr()->toStdString() << endl;
    cout << "\t\tExt: " << newFilePage_4Ptr->getFileExtStrPtr()->toStdString() << endl;


    rideFile->openRdWrFile();
    cout << "HERE" << endl;
    QByteArray tmpBArr;
    tmpBArr.append(editor->toPlainText());
    editor->setPlainText(rideFile->readAll() );
    northTabWidgetPtr->addTab(editor, *newFilePage_4Ptr->getFileNameStrPtr()
                                        + *newFilePage_4Ptr->getFileExtStrPtr());
    
    rideFile->setParallelFileGuiPtr(editor);
    SaveAll::pushToRideFilePtrVec(rideFile);
    
    cout << "Successfully ended file creation sequence" << endl;
    this->close();
    
    FileTreeGui::refresh();
}


void NewFileGui::handleHelpBntSlot()
{
    ;
}


void NewFileGui::handleCancelBtnSlot()
{
    this->close();
}


QAbstractItemModel* NewFileGui::modelFromFile(const QString& fileName)
{
    QFile file(fileName);
    if(!file.open(QFile::ReadOnly))
    {
        cout << "ERROR trying to read file" << endl;
        return new QStringListModel(completer);
    }
    else
    {
        cout << "SUCCESS reading file at NewFileGui::modelFromFile" << endl;
    }
        
    QApplication::setOverrideCursor(QCursor(Qt::WaitCursor));
    QStringList words;

    while (!file.atEnd()) {
        QByteArray line = file.readLine();
        if (!line.isEmpty())
            words << line.trimmed();
    }

    QApplication::restoreOverrideCursor();
    return new QStringListModel(words, completer);
}


void NewFileGui::initBtns()
{
    backBtn = new QPushButton("< Back", this);
    backBtn->setEnabled(false);
    nextBtn = new QPushButton("Next >", this);
    finishBtn = new QPushButton("Finish", this);
    finishBtn->setEnabled(false);
    helpBtn = new QPushButton("Help", this);
    cancelBtn = new QPushButton("Cancel", this);
    
    buttonLayout = new QHBoxLayout();
    buttonLayout->addWidget(backBtn);
    buttonLayout->addWidget(nextBtn);
    buttonLayout->addWidget(finishBtn);
    buttonLayout->addWidget(helpBtn);
    buttonLayout->addWidget(cancelBtn);
    
    connect(backBtn, SIGNAL(released() ), this, SLOT(handleBackBtnSlot() ) );
    connect(nextBtn, SIGNAL(released() ), this, SLOT(handleNextBtnSlot() ) );
    connect(finishBtn, SIGNAL(released() ), this, SLOT(handleFinishBtnSlot() ) );
    connect(helpBtn, SIGNAL(released() ), this, SLOT(handleHelpBntSlot() ) );
    connect(cancelBtn, SIGNAL(released() ), this, SLOT(handleCancelBtnSlot() ) );
    
    outerLayout->addWidget(newFilePage_1Ptr, 0, 0);
    outerLayout->addLayout(buttonLayout, 1, 0, Qt::AlignBottom);
}


void NewFileGui::swapBackPage()
{
    switch(currentPage)
    {
        case PAGE_TWO:
            unloadPage_2();
            loadPage_1();
            currentPage = PAGE_ONE;
            backBtn->setEnabled(false);
            break;
        case PAGE_THREE:
            unloadPage_3();
            loadPage_2();
            currentPage = PAGE_TWO;
            break;
        case PAGE_FOUR:
            unloadPage_4();
            loadPage_3();
            currentPage = PAGE_THREE;
            nextBtn->setEnabled(true);
            finishBtn->setEnabled(false);
            break;
        default:
            cerr << "ERROR in switch at: NewFileGui::swapBackPage()" << endl;
    }
}


void NewFileGui::swapNextPage()
{
    switch(currentPage)
    {
        case PAGE_ONE:
            unloadPage_1();
            loadPage_2();
            currentPage = PAGE_TWO;
            backBtn->setEnabled(true);
            break;
        case PAGE_TWO:
            unloadPage_2();
            loadPage_3();
            currentPage = PAGE_THREE;
            break;
        case PAGE_THREE:
            unloadPage_3();
            loadPage_4();
            currentPage = PAGE_FOUR;
            nextBtn->setEnabled(false);
            finishBtn->setEnabled(true);
            break;
        default:
            cerr << "ERROR in switch at: NewFileGui::swapNextPage()" << endl;
    }
}


void NewFileGui::loadPage_1()
{
    outerLayout->addWidget(newFilePage_1Ptr, 0, 0);
    newFilePage_1Ptr->setVisible(true);
    newFilePage_1Ptr->setEnabled(true);
}


void NewFileGui::unloadPage_1()
{
    outerLayout->removeWidget(newFilePage_1Ptr);
    newFilePage_1Ptr->setVisible(false);
    newFilePage_1Ptr->setEnabled(false);
}


void NewFileGui::loadPage_2()
{
    outerLayout->addWidget(newFilePage_2Ptr, 0, 0);
    newFilePage_2Ptr->setVisible(true);
    newFilePage_2Ptr->setEnabled(true);
}


void NewFileGui::unloadPage_2()
{
    outerLayout->removeWidget(newFilePage_2Ptr);
    newFilePage_2Ptr->setVisible(false);
    newFilePage_2Ptr->setEnabled(false);
}


void NewFileGui::loadPage_3()
{
    outerLayout->addWidget(newFilePage_3Ptr, 0, 0);
    newFilePage_3Ptr->setVisible(true);
    newFilePage_3Ptr->setEnabled(true);
}


void NewFileGui::unloadPage_3()
{
    outerLayout->removeWidget(newFilePage_3Ptr);
    newFilePage_3Ptr->setVisible(false);
    newFilePage_3Ptr->setEnabled(false);
}


void NewFileGui::loadPage_4()
{
    outerLayout->addWidget(newFilePage_4Ptr, 0, 0);
    newFilePage_4Ptr->setVisible(true);
    newFilePage_4Ptr->setEnabled(true);
}


void NewFileGui::unloadPage_4()
{
    outerLayout->removeWidget(newFilePage_4Ptr);
    newFilePage_4Ptr->setVisible(false);
    newFilePage_4Ptr->setEnabled(false);
}


QString* NewFileGui::toString()
{
    QString* tmp = new QString("NewProjectGui");
    tmp->append("\n- - - - - - - - - -\n");
    tmp->append(newFilePage_1Ptr->toString() );
    tmp->append("\n");
    tmp->append(newFilePage_2Ptr->toString() );
    tmp->append("\n");
    tmp->append(newFilePage_3Ptr->toString() );
    tmp->append("\n");
    tmp->append(newFilePage_4Ptr->toString() );
    
    return tmp;
}


NewFileGui::~NewFileGui()
{
    ;
}